R/ingest-scp.R

```diff
@@ -79,45 +77,30 @@ get_input_combinations <- function(cancer_types, race_options, sex_options, age_
 #' )
 #'
 #'
+#'
 
-get_incidence_data <- function(state, chosen_cancer, chosen_race, chosen_sex, chosen_age, chosen_stage, chosen_year){
-  out = as.data.frame(
-    cancerprof::incidence_cancer(
+get_incidence_data <- function(state, cancer, race, sex, age, stage, year){
+  cancerprof::incidence_cancer(
       state, "county",
-      chosen_cancer,
-      chosen_race,
-      chosen_sex,
-      chosen_age,
-      chosen_stage,
-      chosen_year
-    )
-  ) %>%
+      cancer,
+      race,
+      sex,
+      age,
+      stage,
+      year
+    ) %>%
   select(
     County,
     Age_Adjusted_Incidence_Rate
+  ) %>%
+  mutate(
+    Cancer_Type = cancer,
+    Race = race,
+    Sex = sex,
+    Age = age,
+    Stage = stage,
+    Year = year
   )
```

Comment:
To fill a column with a single value, you only need to provide the value once:

```r
library(dplyr)

mtcars |> 
  filter(hp > 200) |> 
  select(mpg, cyl, hp) |> 
  mutate(Fast = TRUE)
```

Also much more readable to do it all in one call to `mutate()`.